lib/runtime/storage/storagediff.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package storage
+
+import (
+	"bytes"
+	"maps"
+	"sync"
+
+	xmaps "golang.org/x/exp/maps"
+
+	"github.com/ChainSafe/gossamer/pkg/trie"
+	"github.com/ChainSafe/gossamer/pkg/trie/codec"
+)
+
+type storageDiff struct {
```

Comment:
I think this would be a case where adding some nice docs in the form of comments would be useful. Maybe just have it be similar to the PR description, describing how this storage diff, batch processing type functionality works.

Reply:
Sure! will do!

Reply:
Done